ex03.exs

```diff
@@ -62,7 +62,35 @@ defmodule Ex03 do
   """
 
   def pmap(collection, process_count, function) do
-    « your code here »
+    collection
+      |> get_chunk_size(process_count)
+      |> divide_work(collection)
+      |> do_parallel_work(function)
+  end
+
+  def get_chunk_size(collection, process_count) do
+    count = Enum.count(collection)
+    calculate_chunk_size(count, process_count, count >= process_count)
+  end
+
+  # number of elements in collection >= supplied process count
+  def calculate_chunk_size(count, process_count, true) do
+    div(count, process_count)
+  end
+
+  # number of elements in collection < supplied process count
+  def calculate_chunk_size(count, _process_count, false) do
+    count
+  end
+
+  def divide_work(chunk_size, collection) do
+    Enum.chunk_every(collection, chunk_size) 
+  end
+
```

Comment:
This is a clever use of `reduce`, but I wonder if it adds anything to the readability of the code over simply piping the map into concat directly?